core/controllers/email_dashboard.py

```diff
@@ -14,6 +14,8 @@
 
 """Controller for user query related pages and handlers."""
 
+from google.appengine.datastore.datastore_query import Cursor
```

Comment:
Confine imports of GAE-related stuff to gae_models and core/platform -- i.e., pass the urlsafe string through and convert it to an actual cursor only in the storage layer. Define an additional helper method in gae_models if necessary.

(This keeps the GAE dependency nicely separated from the rest of the codebase.)